pkg/policy/api/rule.go

```diff
@@ -252,3 +258,50 @@ type PortRuleHTTP struct {
 	// +optional
 	Headers []string `json:"headers,omitempty"`
 }
+
+// PortRuleKafka is a list of Kafka protocol constraints. All fields are
+// optional, if all fields are empty or missing, the rule does not have any
+// effect.
+type PortRuleKafka struct {
+	// APIVersion is the version matched against the api version of the
+	// Kafka message. It is always "0" or a string representing a
+	// positive integer.
+	//
+	// +optional
+	APIVersion string `json:"apiVersion,omitempty"`
+
+	// APIKey is a string matched against the key of a
+	// request, e.g. "Produce", "Fetch", "CreateTopic", "DeleteTopic", ...
+	//
+	//
+	// If omitted or empty, all keys are allowed.
+	//
+	// +optional
+	APIKey string `json:"apiKey,omitempty"`
+
+	// Topic is a regex matched against the topic of the
+	// Kafka message. Ignored if the matched request message type doesn't
+	// contain any topic. Maximum size of Topic can be 249 characters as
+	// per Kafka spec and allowed characters are a-z, A-Z, 0-9, -, . and _
+	//
+	// +optional
+	Topic string `json:"topic,omitempty"`
+}
+
+// KafkaProduceReq, KafkaFetchReq, KafkaCreateTopicsReq, KafkaDeleteTopicsReq
```

Comment:
Why would you limit to a specific subset of API keys? I don’t see any reason why you couldn’t accept all API keys defined in the spec.

Reply:
Changed. We now allow all keys as per spec, in a case-insensitive manner
https://kafka.apache.org/protocol#protocol_api_keys